### PR TITLE
Implement CGConv layer.

### PR DESCRIPTION
**Pull request by emsal0**

### CODE REVIEW

File: docs/src/manual/conv.md [hunk 1]
```diff
@@ -120,3 +120,17 @@ where ``f_{\Theta}`` denotes a neural network parametrized by ``\Theta``, *i.e.*
...
 ```
 Reference: [How Powerful are Graph Neural Networks?](https://arxiv.org/pdf/1810.00826.pdf)
+
+## Crystal Graph Convolutional Network
+
+```math
+\textbf{x}_i' = \textbf{x}_i + \sum_{j \in \mathcal{N}(i)} \sigma\left( \textbf{z}_{i,j} \textbf{W}_f + \textbf{b}_f \right) \odot \text{softplus}\left(\textbf{z}_{i,j} \textbf{W}_s + \textbf{b}_s \right)
+```
+
+where ``\textbf{z}_{i,j} = [\textbf{x}_i, \textbf{x}_j}, \textbf{e}_{i,j}]`` denotes the concatenation of node features, neighboring node features, and edge features.
```
Comment (yuehhua) [inline docs/src/manual/conv.md:130]: should note $\odot$ and $\sigma$ here.

**Comment by yuehhua** [comment]
It seems the same model here [AtomicGraphNets.jl](https://github.com/Chemellia/AtomicGraphNets.jl).

File: src/layers/conv.jl [hunk 2]
```diff
@@ -436,3 +436,77 @@ end
...
+    check_num_nodes(fg, X)
+    check_num_edges(fg, E)
+    _, Y = propagate(c, adjacency_list(fg), E, X, +)
+    Y
+end
+
+(l::CGConv)(fg::FeaturedGraph) = FeaturedGraph(fg, nf=l(fg, node_feature(fg),
+                                                        edge_feature(fg)),
+                                               ef=edge_feature(fg))
+(l::CGConv)(M::AbstractMatrix; edge=false) =
```
Comment (yuehhua) [inline src/layers/conv.jl:506]: The semantics here is a bit ambiguous. `edge` can be replaced as `as_edge` to enhance the meaning of "use `M` as edge feature".

File: src/layers/conv.jl [hunk 3]
```diff
@@ -436,3 +436,77 @@ end
...
+    if !edge
```
Comment (yuehhua) [inline src/layers/conv.jl:507]:
```suggestion
    if edge
```
If it is a bool value, then a negation operation may not needed, just exchange two conditions.

File: docs/src/manual/conv.md (hunk 1 above)
Comment (emsal0) [reply docs/src/manual/conv.md]: Fixec

File: src/layers/conv.jl (hunk 2 above)
Comment (emsal0) [reply src/layers/conv.jl]: Fixed

File: src/layers/conv.jl (hunk 3 above)
Comment (emsal0) [reply src/layers/conv.jl]: Fixed

**Comment by yuehhua** [comment]
Thank you so much!

**Pull request merged**